compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -639,6 +642,8 @@ impl<'hir> LoweringContext<'_, 'hir> {
                 fn_decl,
                 body,
                 fn_decl_span: self.lower_span(span),
+                // FIXME(SarthakSingh31): This span needs to only span the arguments but it doesn't
```

Comment:
This span comes from a desugaring, nothing corresponds to the arguments.

Reply:
Made the `hir::Closure::fn_arg_span` `Option<Span>` to better represent this